packages/language-server/src/__test__/artificial-panic.test.ts

```diff
@@ -0,0 +1,237 @@
+import { CodeActionParams, CompletionParams, DocumentFormattingParams } from 'vscode-languageserver'
+import { TextDocument, TextEdit } from 'vscode-languageserver-textdocument'
+import {
+  handleCodeActions,
+  handleCompletionRequest,
+  handleDiagnosticsRequest,
+  handleDocumentFormatting,
+} from '../MessageHandler'
+import { CURSOR_CHARACTER, findCursorPosition, getTextDocument } from './helper'
+
+import * as assert from 'assert'
+
+suite('Artificial Panics', () => {
+  const OLD_ENV = { ...process.env }
+
+  test('code actions', () => {
+    const fixturePath = './artificial-panic/schema.prisma'
+    const document: TextDocument = getTextDocument(fixturePath)
+
+    const params: CodeActionParams = {
+      textDocument: document,
+      range: {
+        start: { line: 0, character: 0 },
+        end: { line: 0, character: 0 },
+      },
+      context: {
+        diagnostics: [
+          {
+            range: {
+              start: { line: 0, character: 0 },
+              end: { line: 0, character: 0 },
+            },
+            message: 'dx msg',
+          },
+        ],
+      },
+    }
+
+    process.env.FORCE_PANIC_PRISMA_FMT = '1'
+
+    let calledCount = 0
+    let calledArg: undefined | unknown = undefined
+
+    // No official mock implementation in mocha
+    // -> DIY mock for onError
+    const onError = (arg: unknown) => {
+      calledCount += 1
+      calledArg = arg
+    }
+
+    try {
+      const _codeActions = handleCodeActions(params, document, onError)
+
+      assert.fail("This shouldn't happen!")
+    } catch (e) {
+      assert.ok(calledArg)
```

Comment:
Mind that just running `assert.ok(calledArg)` doesn't provide any utility, beyond asserting that we're actually passing a value to the `onError` callback. As discussed previously, we're interested in the content of such value, which should be read by the Wasm panic handler. Please use `assert.strictEqual(calledArg, '....')` or similars

Reply:
This asserts that the value was passed and then also updated. This would fail if it was still undefined

Reply:
Good that you've already manually tested it, now let's automatically test it :)
`assert.ok(calledArg)` as it stands provides little to no additional info w.r.t. `calledCount`.
Now, if we were able to snapshot what kind of error message we pass to the `onError` callback, that would be much more useful, don't you think?

Reply:
> Now, if we were able to snapshot what kind of error message we pass to the onError callback, that would be much more useful, don't you think?

Not really, tbh 😅. I can update it anyway, but I really don't see why; if we're strictly testing the functionality of it, the message can be changed, is that something important to track and update tests for.

Reply:
I now see that the Wasm panic message is only `console.error`'d ([link](https://github.com/prisma/language-tools//blob/057d85f7ddd12326058f899924796a86e9f25070/packages/language-server/src/prisma-fmt/util.ts#L52)), and that `onError()` simply receives a static string.
I recalled that `onError()` received a dynamic panic message from Rust, as I had brought up in past calls / messages.

With that, it's indeed ok to not assert the content the `onError` argument.
Unfortunately `language-tools` doesn't have the `jestContext` utilities we have in `prisma/prisma` to track which arguments were passed to `console.*` methods, so it's impractical to test those (and that's definitely outside the scope of this PR).